scss/_patterns_logo-section.scss

```diff
@@ -0,0 +1,124 @@
+@mixin vf-p-set-nested-logo-width($breakpoint: small) {
+  $grid-column-count: $grid-columns-small;
+  $grid-column-prefix: $grid-small-col-prefix;
+  $gutter: map-get($grid-gutter-widths, small);
+  $media-query-keyword: max-width;
+  $media-query-width: $threshold-4-6-col;
+  $logo-column-span: 1;
+
+  @if $breakpoint == medium {
+    $grid-column-count: $grid-columns-medium;
+    $grid-column-prefix: $grid-medium-col-prefix;
+    $gutter: map-get($grid-gutter-widths, medium);
+    $media-query-keyword: min-width;
+    $media-query-width: $threshold-4-6-col;
+    $logo-column-span: 1;
+  } @else if $breakpoint == large {
+    $grid-column-count: $grid-columns;
+    $grid-column-prefix: $grid-large-col-prefix;
+    $gutter: map-get($grid-gutter-widths, large);
+    $media-query-keyword: min-width;
+    $media-query-width: $threshold-6-12-col;
+    $logo-column-span: 2;
+  }
+
+  @for $i from $grid-column-count through 1 {
+    .#{$grid-column-prefix}#{$i} & {
+      @media ($media-query-keyword: #{$media-query-width}) {
```

Comment:
This is so complicated... 
Is there any way to make it simpler? I know you may have this math in your head now @lyubomir-popov, but will you know in 6 months what was it about? Will anyone else trying to edit it or fix it be able to do it?

I know SCSS is not the nicest programming language to expres such complicated calculations, so maybe at least some comments would help understanding what are we trying to achieve here.

Reply:
The way this loop is written results in tons of very small media querries in the resulting css:

<img width="458" alt="Screenshot 2021-04-06 at 08 38 18" src="https://user-images.githubusercontent.com/83575/113669356-ecbc2e00-96b3-11eb-808a-cfea3285b89f.png">

Would there be a way to rewrite it in a way to group all rules under same media queries?
Like, first do a media query, and then loop though all of the col classes inside it?

Reply:
This calculation in resulting CSS seems wrong, `200% - -2rem`, is this expected?

<img width="260" alt="Screenshot 2021-04-06 at 08 44 02" src="https://user-images.githubusercontent.com/83575/113669619-4c1a3e00-96b4-11eb-807b-2d3d0e246084.png">

Reply:
Was wondering wether to add comments, but it will be 10-15 lines as I go through the steps and then the reordering to ensure clean % - rem value output. I'll do it, see wht you think.

Well spotted about the fragmented media queries, thanks I'll fix that too.